NitroxClient/Unity/Helper/CoroutineHelper.cs

```diff
@@ -0,0 +1,36 @@
+using System;
+using System.Collections;
+
+namespace NitroxClient.Unity.Helper;
+
+public static class CoroutineHelper
+{
+    public static IEnumerator SafelyYieldEnumerator(IEnumerator enumerator, Action<Exception> exceptionCallback)
```

Comment:
Minor: I would turn this into an Extension method because I prefer the look of this:
```cs
yield return UpdateEntities<Base, BuildEntity>(entities.Keys.OfType<BuildEntity>().ToList(), OverwriteBase, (entity, reference) => NitroxVector3.Distance(entity.Transform.LocalPosition, reference) < 0.001f)
            .OnYieldError(exception => Log.Error($"Encountered an exception while resyncing BuildEntities:\n{exception}"));
        yield return UpdateEntities<Constructable, ModuleEntity>(entities.Keys.OfType<ModuleEntity>().ToList(), OverwriteModule, (entity, reference) => NitroxVector3.Distance(entity.Transform.LocalPosition, reference) < 0.001f)
            .OnYieldError(exception => Log.Error($"Encountered an exception while resyncing ModuleEntities:\n{exception}"));
```
Over this:
```cs
yield return CoroutineHelper.SafelyYieldEnumerator(UpdateEntities<Base, BuildEntity>(
            entities.Keys.OfType<BuildEntity>().ToList(), OverwriteBase, (entity, reference) =>
            {
                return NitroxVector3.Distance(entity.Transform.LocalPosition, reference) < 0.001f;
            }
        ), exception => Log.Error($"Encountered an exception while resyncing BuildEntities:\n{exception}"));
        yield return CoroutineHelper.SafelyYieldEnumerator(UpdateEntities<Constructable, ModuleEntity>(
            entities.Keys.OfType<ModuleEntity>().ToList(), OverwriteModule, (entity, reference) =>
            {
                return NitroxVector3.Distance(entity.Transform.LocalPosition, reference) < 0.001f;
            }
        ), exception => Log.Error($"Encountered an exception while resyncing ModuleEntities:\n{exception}"));
```

But YMMV so maybe we should discuss on Discord.

Reply:
I agree this looks way better. Perfect solution because the full name was just too long